source/server/config_validation/server.cc

```diff
@@ -85,7 +85,12 @@ void ValidationInstance::initialize(const Options& options,
   InstanceUtil::loadBootstrapConfig(bootstrap_, options,
                                     messageValidationContext().staticValidationVisitor(), *api_);
 
-  if (!options.logFormatSet() && bootstrap_.has_application_log_format() &&
+  if (options_.logFormatSet() && bootstrap_.has_application_log_format()) {
+    throw EnvoyException(
+        "Only one of application_log_format or CLI option --log-format can be specified.");
+  }
```

Comment:
nit: I would factor here and below into some tiny shared helper just so we share the same logic and error message in case it changes in the future.

Reply:
done